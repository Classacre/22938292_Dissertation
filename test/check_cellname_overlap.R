# Check cell name matching between expression matrix and metadata
expr_mat <- as.matrix(read.csv("/group/sms029/mnieuwenh/gbm_noise_analysis/expression_matrix.csv", row.names=1, check.names=FALSE))
cell_metadata <- read.csv("/group/sms029/mnieuwenh/seurat_metadata/seurat_metadata_sample.csv", row.names=1, check.names=FALSE)

cat("First 5 colnames in expression matrix:\n")
print(head(colnames(expr_mat), 5))
cat("\nFirst 5 rownames in metadata:\n")
print(head(rownames(cell_metadata), 5))

cat("\nNumber of overlapping cell names:\n")
print(length(intersect(colnames(expr_mat), rownames(cell_metadata))))

cat("\nExample of non-overlapping cell names in expr_mat:\n")
print(setdiff(colnames(expr_mat), rownames(cell_metadata))[1:5])
cat("\nExample of non-overlapping cell names in metadata:\n")
print(setdiff(rownames(cell_metadata), colnames(expr_mat))[1:5])
