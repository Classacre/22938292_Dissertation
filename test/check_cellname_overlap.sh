#!/bin/bash
#SBATCH --job-name=check_cellname_overlap
#SBATCH --output=/group/sms029/mnieuwenh/test/logs/check_cellname_overlap.out
#SBATCH --error=/group/sms029/mnieuwenh/test/logs/check_cellname_overlap.err
#SBATCH --partition=work
#SBATCH --mem=16G
#SBATCH --cpus-per-task=1
#SBATCH --time=01:00:00
#SBATCH --mail-user=22938292@student.uwa.edu.au
#SBATCH --mail-type=END,FAIL

module load Anaconda3/2024.06
source ~/.bashrc
conda activate /group/sms029/conda_environment/R

Rscript /group/sms029/mnieuwenh/test/check_cellname_overlap.R
