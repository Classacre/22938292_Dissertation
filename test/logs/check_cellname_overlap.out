First 5 colnames in expression matrix:
[1] "V1" "V2" "V3" "V4" "V5"

First 5 rownames in metadata:
[1] "AAACCTGAGAGCCCAA-1_1" "AAACCTGAGGCCCTTG-1_1" "AAACGGGAGCTAGGCA-1_1"
[4] "AAACGGGCACCCTATC-1_1" "AAACGGGCAGCATACT-1_1"

Number of overlapping cell names:
[1] 0

Example of non-overlapping cell names in expr_mat:
[1] "V1" "V2" "V3" "V4" "V5"

Example of non-overlapping cell names in metadata:
[1] "AAACCTGAGAGCCCAA-1_1" "AAACCTGAGGCCCTTG-1_1" "AAACGGGAGCTAGGCA-1_1"
[4] "AAACGGGCACCCTATC-1_1" "AAACGGGCAGCATACT-1_1"
